src/containers/sourcecast/SourcecastContainer.ts

```diff
@@ -0,0 +1,117 @@
+  replValue: state.workspaces.sourcecast.replValue,
+  sideContentHeight: state.workspaces.sourcecast.sideContentHeight,
+  sourcecastIndex: state.workspaces.sourcecast.sourcecastIndex,
+  sourceChapter: state.workspaces.sourcecast.context.chapter,
+  websocketStatus: state.workspaces.playground.websocketStatus
+});
+
+const location: WorkspaceLocation = 'sourcecast';
+
+const mapDispatchToProps: MapDispatchToProps<IDispatchProps, {}> = (dispatch: Dispatch<any>) =>
+  bindActionCreators(
+    {
+      handleBrowseHistoryDown: () => browseReplHistoryDown(location),
+      handleBrowseHistoryUp: () => browseReplHistoryUp(location),
+      handleChangeActiveTab: (activeTab: number) => changeActiveTab(activeTab, location),
+      handleChapterSelect: (chapter: number) => chapterSelect(chapter, location),
+      handleEditorEval: () => evalEditor(location),
+      handleEditorValueChange: (val: string) => updateEditorValue(val, location),
+      handleEditorHeightChange: (height: number) => changeEditorHeight(height, location),
+      handleEditorWidthChange: (widthChange: number) => changeEditorWidth(widthChange, location),
+      handleEditorUpdateBreakpoints: (breakpoints: string[]) =>
+        setEditorBreakpoint(breakpoints, location),
+      handleFetchSourcecastIndex: () => fetchSourcecastIndex(location),
+      handleInterruptEval: () => beginInterruptExecution(location),
+      handleReplEval: () => evalRepl(location),
+      handleReplOutputClear: () => clearReplOutput(location),
+      handleReplValueChange: (newValue: string) => updateReplValue(newValue, location),
+      handleSetCodeDeltasToApply: (deltas: ICodeDelta[]) => setCodeDeltasToApply(deltas, location),
+      handleSetEditorReadonly: (editorReadonly: boolean) =>
+        setEditorReadonly(location, editorReadonly),
+      handleSetInputToApply: (inputToApply: Input) => setInputToApply(inputToApply, location),
+      handleSetSourcecastData: (
+        title: string,
+        description: string,
+        audioUrl: string,
+        playbackData: IPlaybackData
+      ) => setSourcecastData(title, description, audioUrl, playbackData, location),
+      handleSetSourcecastDuration: (duration: number) => setSourcecastDuration(duration, location),
+      handleSetSourcecastStatus: (playbackStatus: PlaybackStatus) =>
+        setSourcecastStatus(playbackStatus, location),
+      handleSetWebsocketStatus: (websocketStatus: number) =>
+        setWebsocketStatus(location, websocketStatus),
+      handleSideContentHeightChange: (heightChange: number) =>
+        changeSideContentHeight(heightChange, location),
+      handleToggleEditorAutorun: () => toggleEditorAutorun(location),
+      handleDebuggerPause: () => beginDebuggerPause(location),
+      handleDebuggerResume: () => debuggerResume(location),
+      handleDebuggerReset: () => debuggerReset(location)
+    },
+    dispatch
+  );
+
+export default connect(
+  mapStateToProps,
+  mapDispatchToProps
+)(Sourcecast);
```

Comment:
Is it possible to practice DRY here. We already have a connected `workspace`, and `Sourcecast` renders that `workspace`. Using composition, we should only add new mappings related to sourcecast here.

Reply:
> Is it possible to practice DRY here. We already have a connected `workspace`, and `Sourcecast` renders that `workspace`. Using composition, we should only add new mappings related to sourcecast here.

Not sure what you mean by we already have a connected `workspace`?
As far as I can tell `workspace` is just a component by itself and is not connected to the redux store.